mongo/datadog_checks/mongo/api.py

```diff
@@ -111,3 +115,23 @@ def _get_deployment_type(self):
             return ReplicaSetDeployment(replset_name, replset_state, cluster_role=cluster_role)
 
         return StandaloneDeployment()
+
+    def _get_alibaba_deployment_type(self):
+        is_master_payload = self['admin'].command('isMaster')
```

Comment:
Are we able to save the value/response from `__init__` and reuse here?
https://github.com/DataDog/integrations-core/blob/4a5e1fd604206d9b13b5ab35dd69b9f04b6d63ed/mongo/datadog_checks/mongo/api.py#L41

Reply:
We could be saving it but performance does not require it. This command returns instantly. Also `_get_alibaba_deployment_type` is called on every check run so we would have to handle a cache with a TTL. Definitely not worth it I believe